src/lib.rs

```diff
@@ -0,0 +1,230 @@
+#![cfg_attr(test, feature(test))]
+
+use indicatif::{ProgressBar, ProgressStyle};
+use linkify::{LinkFinder, LinkKind};
+use rand::{Rng, RngCore, SeedableRng};
+use rand_xoshiro::{Xoshiro256Plus, Xoshiro256PlusPlus};
+use std::fs::File;
+use std::io::{BufWriter, Error, Write};
+use std::path::Path;
+
+use constants::ACTIONS;
+use constants::ACTIONS_SIZE;
+use constants::FACES;
+use constants::FACES_SIZE;
+
+mod constants;
+
+macro_rules! progress_bar {
+    () => {{
+        let progress_bar = ProgressBar::new_spinner();
+        progress_bar.set_style(
+            ProgressStyle::default_spinner().template("{spinner:.green} [{elapsed_precise}] {msg}"),
+        );
+        progress_bar.enable_steady_tick(100);
+
+        progress_bar
+    }};
+}
+
+#[derive(Debug)]
+pub struct UwUify<'a> {
```

Comment:
is the `'a` lifetime here equivalent to `'static` ?

Reply:
No. The `'a` lifetime is equivalent to the lifetime of the struct itself
```rust
fn main() {
    let matches = app!().get_matches();

    match UwUify::new(
        matches.value_of("text"),
        matches.value_of("infile"),
        matches.value_of("outfile"),
        matches.value_of("words"),
        matches.value_of("faces"),
        matches.value_of("actions"),
        matches.value_of("stutters"),
        matches.is_present("random"),
    )
    .uwuify()
    {
        Ok(_) => (),
        Err(err) => {
            app!().error(ErrorKind::DisplayHelp, err).exit();
        }
    }
}
```
Keep in mind that the value of `app!().get_matches()` cannot be known at compile time, therefore using values produced by it under the `'static` lifetime would be invalid